datafusion/proto/src/logical_plan.rs

```diff
@@ -605,6 +605,11 @@ impl AsLogicalPlan for LogicalPlanNode {
                         join.join_constraint
                     ))
                 })?;
+                let filter: Option<Expr> = join
+                    .filter
+                    .as_ref()
+                    .map(|expr| parse_expr(expr, ctx))
+                    .map_or(Ok(None), |v| v.map(Some))?;
```

Comment:
Maybe https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose ? I'm having a hard time figuring out what the types are though so could be mistaken

Reply:
Transpose seems to be appropriate -- I used this map_or call in ballista for conversion from Option<Result> to Result<Option>

Reply:
I copied the code over from Ballista and am not too familiar with what is going on here. @korowa would you mind updating this in a follow on PR?